simplified-cardcreator/src/main/java/org/nypl/simplified/cardcreator/model/JuvenilePatron.kt

```diff
@@ -0,0 +1,19 @@
+package org.nypl.simplified.cardcreator.model
+
+import com.squareup.moshi.Json
+
+sealed class JuvenilePatron
+
+data class UsernameParent(
+  @field:Json(name = "name") val name: String,
+  @field:Json(name = "parentUsername") val parentUsername: String,
+  @field:Json(name = "username") val username: String,
+  @field:Json(name = "pin") val pin: String
+) : JuvenilePatron()
+
+data class BarcodeParent(
+  @field:Json(name = "barcode") val barcode: String,
+  @field:Json(name = "name") val name: String,
+  @field:Json(name = "username") val username: String,
+  @field:Json(name = "pin") val pin: String
```

Comment:
I think you can omit these field annotations since the serialized name is the same as the data class member. Would make it more readable.